tests/test_packages/test_protocols/test_tendermint.py

```diff
@@ -0,0 +1,183 @@
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains tests for the register protocol."""
+
+import itertools
+import logging
+from typing import Type
+from unittest import mock
+
+import pytest
+
+from aea.common import Address
+from aea.mail.base import Envelope
+from aea.protocols.base import Message
+from aea.protocols.dialogue.base import Dialogue as BaseDialogue
+from aea.protocols.dialogue.base import DialogueLabel
+
+from packages.valory.protocols.tendermint.dialogues import (
+    TendermintDialogue,
+    TendermintDialogues as BaseTendermintDialogues,
+)
+from packages.valory.protocols.tendermint.message import TendermintMessage
+
+PERFORMATIVE = TendermintMessage.Performative
+
+ALICE_TENDERMINT_INFO = dict(ip="192.168.0.0", hostname="alice")
+BOB_TENDERMINT_INFO = dict(ip="192.168.255.255", hostname="bob")
+
+
+def test_performative_name_value_matching():
+    """Test performative name, value, and match."""
+    for performative in TendermintMessage.Performative:
+        assert performative.name.isupper()
+        assert performative.value.islower()
+        assert performative.name == performative.value.upper()
+
+
+@pytest.mark.parametrize(
+    "performative, kwargs",
+    [
+        (PERFORMATIVE.TENDERMINT_CONFIG_REQUEST, dict(query="")),
+        (PERFORMATIVE.TENDERMINT_CONFIG_RESPONSE, dict(info=ALICE_TENDERMINT_INFO)),
+        # (TendermintMessage.Performative.ERROR, dict(
+        #     error_code=TendermintMessage.ErrorCode.INVALID_REQUEST,
+        #     error_msg="Invalid request",
+        #     info={"error": "Agent address not registered for this service."}
+        # )),
+    ],
+)
+def test_serialization(performative, kwargs):
+    """Test that the serialization for the 'tendermint' protocol works."""
+
+    msg = TendermintMessage(
+        message_id=1,
+        dialogue_reference=(str(0), ""),
+        target=0,
+        performative=performative,
+        **kwargs
+    )
+
+    msg.to = "receiver"
+    original_envelope = Envelope(to=msg.to, sender="sender", message=msg)
+    retrieved_envelope = Envelope.decode(original_envelope.encode())
+    # assert original_envelope = retrieved_envelope
+    assert retrieved_envelope.to == original_envelope.to
+    assert retrieved_envelope.sender == original_envelope.sender
+    assert (
+        retrieved_envelope.protocol_specification_id
+        == original_envelope.protocol_specification_id
+    )
+    assert retrieved_envelope.sender == original_envelope.sender
+    assert retrieved_envelope.context == original_envelope.context
+
+    # what? why don't we decode the message contained in the envelope?
+    logging.error(original_envelope.context)
+    logging.error(retrieved_envelope.context)
```

Comment:
this seems odd to me, will look a little closer tmrw and then decide to raise an issue on this or not